utils/src/structs.rs

```diff
@@ -130,3 +168,71 @@ impl Job {
         &self.input
     }
 }
+
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub enum Permission {
+    ProcRead,
+    AllExtensions,
+    Extension(String),
+    AllHttpHosts,
+    HttpHost(String),
+}
+
+impl Display for Permission {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let str = match self {
+            Permission::ProcRead => String::from("proc:read"),
+            Permission::AllExtensions => String::from("extension:*"),
+            Permission::Extension(name) => format!("extension:{name}"),
+            Permission::AllHttpHosts => String::from("http:*"),
```

Comment:
Reading `http:` somewhere makes me go "but where is `https:`"?
Does this actually just support HTTP, or does it actually support arbitrary TCP/UDP traffic? In which case, how about `web:` or `connection:` or ...?

Reply:
It should support HTTP requests of any kind, including unencrypted, TLS, http/2, http/3, you name it. Try not to read `http:` as the first part of a URL and try to read it as a namespace for the permissions instead.

Reply:
I'm certainly trying but my stupid brain always goes "where's the slash slash after the colon" 😂